### PR TITLE
Add support for authentication logs API

### PR DESCRIPTION
**Pull request by bgarciaentornos**
NOTE: Official documentation (https://canvas.instructure.com/doc/api/authentications_log.html) is not accurate

### CODE REVIEW

File: src/main/java/edu/ksu/canvas/model/AuthenticationLog.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+        public void setLinks(AuthenticationEventLinks links) {
+            this.links = links;
+        }
+
+        public class AuthenticationEventLinks {
+
+            private Integer user;
+            private Integer account;
+            private Integer login;
```
Comment (buckett) [inline src/main/java/edu/ksu/canvas/model/AuthenticationLog.java:65]: These should all be Long, in some situations Canvas will return full IDs that overflow Integers.

File: src/main/java/edu/ksu/canvas/model/AuthenticationLog.java [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+    public List<AuthenticationEvent> getEvents() {
+        return events;
+    }
+
+    public void setEvents(List<AuthenticationEvent> events) {
+        this.events = events;
+    }
+
+    public Object getLinked() {
```
Comment (buckett) [inline src/main/java/edu/ksu/canvas/model/AuthenticationLog.java:102]: Could you add //TODOs on the Objects.

**Review by buckett — COMMENTED** [review]
Looks good.
I think a issue should be raised with Instructure that their documentation is wrong.

**Comment by bgarciaentornos** [comment]
I have applied the proposed changes and have told Nick to open an issue with Instructure about the documentation (I can't do it myself).

**Review by buckett — APPROVED** [review]

**Pull request merged**